x-pack/plugins/dataset_quality/public/components/log_explorer_link.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLink } from '@elastic/eui';
+import React from 'react';
+import { getRouterLinkProps } from '@kbn/router-utils';
+import {
+  SingleDatasetLocatorParams,
+  SINGLE_DATASET_LOCATOR_ID,
+} from '@kbn/deeplinks-observability';
+import { DataStreamStat } from '../../common/data_streams_stats/data_stream_stat';
+import { useKibanaContextForPlugin } from '../utils';
+
+export const LogExplorerLink = React.memo(
+  ({ dataStreamStat, title }: { dataStreamStat: DataStreamStat; title: string }) => {
+    const {
+      services: { share },
+    } = useKibanaContextForPlugin();
+    const [dataset, namespace] = dataStreamStat.title.split('-');
```

Comment:
**suggestion:** this logic seems repeated in the `DataStreamStat` class definition, we could abstract it into a method that also takes care of edge cases.

Reply:
I will address this one in https://github.com/elastic/kibana/pull/173087, where we have already a dedicated property for `namespace`.